pymeasure/instruments/hp/hp437b.py

```diff
@@ -0,0 +1,949 @@
+        self.write("EX")
+        return float(display_content[7:12])
+
+    @calibration_factor.setter
+    def calibration_factor(self, calibration_factor):
+        values = [1.0, 150.0]
+        strict_range(float(calibration_factor), values)
+
+        self.write("KB%3.1fPCT" % float(calibration_factor))
+        self.check_errors()
+
+    display_enabled = Instrument.setting(
+        "%s",
+        """
+        Set the display of the power meter active or inactive.
+        """,
+        map_values=True,
+        values={True: "DE", False: "DD"}
+    )
+
+    display_all_segments_enabled = Instrument.setting(
+        "%s",
+        """
+        Set all segments of the display of the power meter active or resume normal state.
+        """,
+        map_values=True,
+        values={True: "DA", False: "DE"}
+    )
+
+    display_user_message = Instrument.setting(
+        "DU %s",
+        """
+        Set a custom user message up to 12 alpha-numerical chars. If the string is empty or None
+        the user message gets disabled.
+        """,
+        validator=lambda x, y: x if str(x).isalnum() and len(str(x)) <= 12 else str(x)[0:11],
+        set_process=lambda v: str(v).upper().ljust(12)
+    )
+
+    display_output = Instrument.measurement(
+        "OD",
+        """
+        Get the current displayed string of values of the power meter.
+
+        .. code-block:: python
+
+            print(instr.display_output)
+            -0.23  dB REL
+
+        """,
+        cast=str
+    )
+
+    duty_cycle_enabled = Instrument.control(
+        "SM", "DC%d",
+        """
+        Set duty cycle active or inactive. See :attr:`duty_cycle`
+        """,
+        map_values=True,
+        values={True: 1, False: 0},
+        cast=int,
+        get_process=__getstatus(StatusMessage.DutyCycleStatus),
+        check_set_errors=True
+    )
+
+    @property
+    def duty_cycle(self):
+        """
+        Control the duty cycle for calculation of a pulsed input signal. This function will cause
+        the power meter to report the pulse power of a rectangular pulsed input signal. The
+        allowable range of values for 'duty_cycle' is 0.001 to 99.999%.
+
+        Pulse power, as reported by the power meter, is a mathematical
+        representation of the pulse power rather than an actual measurement.
+        The power meter measures the average power of the pulsed input
+        signal and then divides the measurement by the duty cycle value to
+        obtain a pulse power reading.
+        """
+        self.write("DY")
+        # returns DTYCY 01.000%
+        display_content = self.display_output
+        assert display_content[0:5] == "DTYCY"
+        self.write("EX")
+        return float(display_content[6:12])
+
+    @duty_cycle.setter
+    def duty_cycle(self, duty_cycle):
+        values = [0.001, 99.999]
+        strict_range(float(duty_cycle), values)
+
+        self.write("DY%02.3fPCT" % float(duty_cycle))
+        self.check_errors()
```

Comment:
While the documentation talks of a duty cycle in %, wouldn't it make more sense to represent the value of the dutycycle in python as an absolute figure?
Instead of 5 (%) --> 0.05 or 23.45 (%) --> 0.2345

Reply:
I agree